extensions/notebook/src/book/bookPinManager.ts

```diff
@@ -58,7 +58,8 @@ export class BookPinManager implements IBookPinManager {
 			pinnedBooks.splice(existingBookIndex, 1);
 			modifiedPinnedBooks = true;
 		} else if (existingBookIndex === -1 && operation === PinBookOperation.Pin) {
-			let addNotebook: IPinnedNotebook = { notebookPath: bookPathToChange, bookPath: notebook.book.root, title: notebook.book.title };
+			const bookPath = notebook.tableOfContents.sections ? notebook.book.root : undefined;
```

Comment:
if we're not using this bookPath anywhere else, just add the undefined check inline.